File: algorithm/stack2.py
# https://www.acmicpc.net/problem/10799
# 실버2
galho=input()
stack=[]
answer=0

for i in range(len(galho)):
    if galho[i] == '(':
        stack.append(galho[i])
    else:
        if galho[i-1] == '(':
            stack.pop()
            answer += len(stack)
        else:
            stack.pop()
            answer += 1

print(answer)

File: algorithm-js/stack2.js
// 실버2 https://www.acmicpc.net/problem/10799
const input = require('fs').readFileSync('/dev/stdin').toString().trim();

const stickWithRazer = input.split('');
const stack = [];
let count = 0;

for (let i = 0; i < stickWithRazer.length; i++) {
    if (stickWithRazer[i] === '(') {
        stack.push(stickWithRazer[i]);
    } else {
        if (stickWithRazer[i - 1] === '(') {
            stack.pop();
            count += stack.length;
        } else {
            stack.pop();
            count += 1;
        }
    }
}
console.log(count);